fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/ReferencesRdfContext.java

```diff
@@ -48,21 +57,36 @@ public ReferencesRdfContext(final FedoraResource resource,
         throws RepositoryException {
         super(resource, idTranslator);
         property2triple = new PropertyToTriple(resource.getNode().getSession(), idTranslator);
-        concat(putStrongReferencePropertiesIntoContext());
-        concat(putWeakReferencePropertiesIntoContext());
+        session = resource.getNode().getSession();
+        putReferencesIntoContext(resource.getNode().getWeakReferences());
+        putReferencesIntoContext(resource.getNode().getReferences());
     }
 
-    private Iterator<Triple> putWeakReferencePropertiesIntoContext() throws RepositoryException {
-        final Iterator<Property> properties = resource().getNode().getWeakReferences();
+    private void putReferencesIntoContext(final Iterator<Property> properties) throws RepositoryException {
+        while (properties.hasNext()) {
+            final Property p = properties.next();
+            concat(property2triple.apply(p));
 
-        return Iterators.concat(Iterators.transform(properties, property2triple));
+            for ( final PropertyIterator it = p.getParent().getProperties(); it.hasNext(); ) {
+                final Property potentialProxy = it.nextProperty();
+                if (potentialProxy.isMultiple()) {
+                    for ( Value v : potentialProxy.getValues() ) {
+                        putProxyReferencesIntoContext(v);
+                    }
+                } else {
+                    putProxyReferencesIntoContext(potentialProxy.getValue());
+                }
+            }
+        }
     }
-
-    private Iterator<Triple> putStrongReferencePropertiesIntoContext() throws RepositoryException {
-        final Iterator<Property> properties = resource().getNode().getReferences();
-
-        return Iterators.concat(Iterators.transform(properties, property2triple));
-
+    private void putProxyReferencesIntoContext(final Value v) throws RepositoryException {
+        if (v.getType() == PATH) {
+            putProxyReferencesIntoContext(session.getNode(v.getString()));
+        } else if (v.getType() == REFERENCE || v.getType() == WEAKREFERENCE) {
+            putProxyReferencesIntoContext(session.getNodeByIdentifier(v.getString()));
+        }
+    }
+    private void putProxyReferencesIntoContext(final Node n) throws RepositoryException {
+        concat(new LdpContainerRdfContext(nodeConverter.convert(n), translator()));
```

Comment:
Performing manual testing (see comment in ticket: https://jira.duraspace.org/browse/FCREPO-1497) I was not able to demonstrate the desired behavior. 
That fact withstanding, I am vaguely concerned about potentially adding "LdpContainerRdfContexts" for all references coming off of incoming-reference nodes... but will check more deeply once I can see the desired behavior of this ticket.